### PR TITLE
Move TestExporter to avoid requiring pytest

### PR DESCRIPTION
**Pull request by alexmojaki**
Closes https://github.com/pydantic/logfire/issues/367

### CODE REVIEW

File: logfire/_internal/config.py [hunk 1]
```diff
@@ -75,6 +74,7 @@
 from .exporters.quiet_metrics import QuietMetricExporter
 from .exporters.remove_pending import RemovePendingSpansExporter
 from .exporters.tail_sampling import TailSamplingOptions, TailSamplingProcessor
+from .exporters.test import TestExporter
```
Comment (alexmojaki) [inline logfire/_internal/config.py:77]: @Kludex it's not critical but I don't know why there isn't a stub for `exporters.test` when I run `rye run generate-stubs`. There's other stubs in the `exporters` folder.

Comment (Kludex) [reply logfire/_internal/config.py:77]: I think it's because the `testing` module is not in the `__init__.py`.

Comment (alexmojaki) [reply logfire/_internal/config.py:77]: I'm talking about `logfire._internal.exporters.test`, not `logfire.testing`

**Review by Kludex — APPROVED** [review]

File: logfire/_internal/config.py (hunk 1 above)
Comment (Kludex) [reply logfire/_internal/config.py:77]: Ah, because we don't include private objects.

Comment (Kludex) [reply logfire/_internal/config.py:77]: We'd need to add `--include-private` on the `stubgen` command for it to appear, I think...

Comment (alexmojaki) [reply logfire/_internal/config.py:77]: But all the other internal exporter files are there

**Pull request merged**